lib/renderer/api/desktop-capturer.js

```diff
@@ -27,7 +27,39 @@ function mapSources (sources) {
   }))
 }
 
-exports.getSources = function (options, callback) {
+exports.getSources = (options) => {
+  return new Promise((resolve, reject) => {
+    if (!isValid(options)) {
+      reject(new Error('Invalid options'))
```

Comment:
`throw new Error('Invalid options')` instead, this will reject the promise and prevent the rest of the code from executing